plugins/modules/zos_script.py

```diff
@@ -0,0 +1,416 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+---
+module: zos_script
+version_added: '1.8.0'
+author:
+  - "Ivan Moreno (@rexemin)"
+short_description: Run scripts in z/OS
+description:
+  - The L(zos_script,./zos_script.html) module runs a local or remote script
+    in the remote machine.
+
+options:
+  chdir:
+    description:
+      - Change the script's working directory to this path.
+      - When not specified, the script will run in the user's
+        home directory on the remote machine.
+    type: str
+    required: false
+  cmd:
+    description:
+      - Path to the local or remote script followed by optional arguments.
+      - If the script path contains spaces, make sure to enclose it in two
+        pairs of quotes.
+      - Arguments may need to be escaped so the shell in the remote machine
+        handles them correctly.
+    type: str
+    required: true
+  creates:
+    description:
+      - Path to a file in the remote machine. If it exists, the
+        script will not be executed.
+    type: str
+    required: false
+  encoding:
+    description:
+      - Specifies which encodings the script should be converted from and to.
+      - If C(encoding) is not provided, the module determines which local
+        and remote charsets to convert the data from and to.
+    type: dict
+    required: false
+    suboptions:
+      from:
+        description:
+          - The encoding to be converted from.
+        required: true
+        type: str
+      to:
+        description:
+          - The encoding to be converted to.
+        required: true
+        type: str
+  executable:
+    description:
+      - Path of an executable in the remote machine to invoke the
+        script with.
+      - When not specified, the system will assume the script is
+        interpreted REXX and try to run it as such. Make sure to
+        include a comment identifying the script as REXX at the
+        start of the file in this case.
+    type: str
+    required: false
+  remote_src:
+    description:
+      - If set to C(false), the module will search the script in the
+        controller.
+      - If set to C(true), the module will search the script in the
+        remote machine.
+    type: bool
+    required: false
+  removes:
+    description:
+      - Path to a file in the remote machine. If it does not exist, the
+        script will not be executed.
+    type: str
+    required: false
+  tmp_path:
```

Comment:
We should use the Ansible variable, that would probably be for the better, it would be easier for other modules to adopt it, unless users really need to be able to set a temp path at the task level. We could try leveraging the Ansible option on the GA